Compiler warning: call 'std::move' explicitly to avoid copying

```bash
In file included from /usr/local/include/xtensor/xarray.hpp:19:
In file included from /usr/local/include/xtensor/xcontainer.hpp:23:
In file included from /usr/local/include/xtensor/xmath.hpp:26:
In file included from /usr/local/include/xtensor/xoperation.hpp:21:
In file included from /usr/local/include/xtensor/xstrided_view.hpp:26:
In file included from /usr/local/include/xtensor/xstrided_view_base.hpp:18:
/usr/local/include/xtensor/xslice.hpp:903:24: warning: local variable 'slice' will be copied despite being returned by name [-Wreturn-std-move]
                return slice;
                       ^~~~~
/usr/local/include/xtensor/xslice.hpp:960:16: note: in instantiation of function template specialization 'xt::detail::keep_drop_getter::operator()<const xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag>, xt::xkeep_slice<unsigned long> >' requested here
        return getter(e, std::forward<SL>(slice), index);
               ^
/usr/local/include/xtensor/xview.hpp:1607:17: note: in instantiation of function template specialization 'xt::get_slice_implementation<const xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag>, xt::xkeep_slice<unsigned long> >' requested here
                get_slice_implementation(e, std::forward<S>(slices), get_underlying_shape_index<std::decay_t<E>, S...>(I))...
                ^
/usr/local/include/xtensor/xview.hpp:1623:24: note: in instantiation of function template specialization 'xt::detail::make_view_impl<const xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag> &, 0, 1, xt::xkeep_slice<unsigned long>, xt::xall_tag>' requested here
        return detail::make_view_impl(std::forward<E>(e), std::make_index_sequence<sizeof...(S)>(), std::forward<S>(slices)...);
                       ^
/usr/local/include/GooseFEM/TyingsPeriodic.hpp:50:47: note: in instantiation of function template specialization 'xt::view<const xt::xtensor_container<xt::uvector<unsigned long, xsimd::aligned_allocator<unsigned
      long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag> &, xt::xkeep_slice<unsigned long>, xt::xall_tag>' requested here
  xt::xtensor<size_t,1> iid = xt::flatten(xt::view(dofs, xt::keep(dependent), xt::all()));
                                              ^
/usr/local/include/xtensor/xslice.hpp:903:24: note: call 'std::move' explicitly to avoid copying
                return slice;
                       ^~~~~
                       std::move(slice)
/usr/local/include/xtensor/xslice.hpp:903:24: warning: local variable 'slice' will be copied despite being returned by name [-Wreturn-std-move]
                return slice;
                       ^~~~~
/usr/local/include/xtensor/xslice.hpp:960:16: note: in instantiation of function template specialization 'xt::detail::keep_drop_getter::operator()<xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag>, xt::xkeep_slice<long> >' requested here
        return getter(e, std::forward<SL>(slice), index);
               ^
/usr/local/include/xtensor/xview.hpp:1607:17: note: in instantiation of function template specialization 'xt::get_slice_implementation<xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag>, xt::xkeep_slice<long> >' requested here
                get_slice_implementation(e, std::forward<S>(slices), get_underlying_shape_index<std::decay_t<E>, S...>(I))...
                ^
/usr/local/include/xtensor/xview.hpp:1623:24: note: in instantiation of function template specialization 'xt::detail::make_view_impl<xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag> &, 0, 1, xt::xkeep_slice<long>, xt::xall_tag>' requested here
        return detail::make_view_impl(std::forward<E>(e), std::make_index_sequence<sizeof...(S)>(), std::forward<S>(slices)...);
                       ^
/data/prog/src/GooseFEM/docs/examples/Statics/LinearElasticityPeriodic/main.cpp:46:26: note: in instantiation of function template specialization 'xt::view<xt::xtensor_container<xt::uvector<unsigned long,
      xsimd::aligned_allocator<unsigned long, 32> >, 2, xt::layout_type::row_major, xt::xtensor_expression_tag> &, xt::xkeep_slice<long>, xt::xall_tag>' requested here
    xt::reshape_view(xt::view(dofs, xt::keep(mesh.nodesOrigin()), xt::all()), {ndim})
                         ^
/usr/local/include/xtensor/xslice.hpp:903:24: note: call 'std::move' explicitly to avoid copying
                return slice;
                       ^~~~~
                       std::move(slice)
```

With 

```bash
Apple LLVM version 10.0.1 (clang-1001.0.46.3)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

--- a/include/xtensor/xslice.hpp
+++ b/include/xtensor/xslice.hpp
@@ -888,19 +888,19 @@ namespace xt
         struct slice_implementation_getter
         {
             template <class E, class SL>
-            inline auto operator()(E&, SL&& slice, std::size_t) const
+            inline decltype(auto) operator()(E&, SL&& slice, std::size_t) const
             {
-                return slice;
+                return std::forward<SL>(slice);
             }
         };
 
         struct keep_drop_getter
         {
             template <class E, class SL>
-            inline auto operator()(E& e, SL&& slice, std::size_t index) const
+            inline decltype(auto) operator()(E& e, SL&& slice, std::size_t index) const
             {
                 slice.normalize(e.shape()[index]);
-                return slice;
+                return std::forward<SL>(slice);
             }
 
             template <class E, class SL>

